// @shopify/cli/dist/node-package-manager-URP7ZOL6.js
import {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForCachedNewVersion,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  inferPackageManager,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  lockfilesByManager,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  versionSatisfies,
  writePackageJSON,
  yarnLockfile
} from "./chunk-JAKMNE2C.js";
import "./chunk-T3I7QQ2S.js";
import "./chunk-HMDWNGIV.js";
import "./chunk-B5EXYCV3.js";
import "./chunk-75LV6AQS.js";
import "./chunk-57Q6OQG4.js";
import "./chunk-UTDSSCPI.js";
import "./chunk-3I3GQNEW.js";
import "./chunk-G5R6YD27.js";
import "./chunk-G2ZZKGSV.js";
import "./chunk-MP5RLDFN.js";
import "./chunk-XE5EOEBL.js";
import "./chunk-5DBS6QOU.js";
import "./chunk-B3YWYURY.js";
import "./chunk-CDBXAE2F.js";
import "./chunk-PKR7KJ6P.js";
export {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForCachedNewVersion,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  inferPackageManager,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  lockfilesByManager,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  versionSatisfies,
  writePackageJSON,
  yarnLockfile
};
